unityfiles/Assets/EnemyHealthBarGUIController.cs

```diff
@@ -0,0 +1,45 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+/// <summary>
+/// Takes care of displaying and updating the enemy's health bar
+/// </summary>
+public class EnemyHealthBarGUIController : MonoBehaviour {
+    
+    [SerializeField]
+    private GameObject healthBarBackground;
+
+    private float startHealth;
+
+    [SerializeField]
+    private Image healthBar;
+
+
+    public void SetStartHealth(float value) {
+        startHealth = value;
+    }
+
+    // Start is called before the first frame update
+    void Start() {
+        HideHealthBar();
+    }
+
+    private void HideHealthBar() {
+        healthBarBackground.SetActive(false);
+    }
+
+    private void DisplayHealthBar() {
+        healthBarBackground.SetActive(true);
+    }
+
+    /// <summary>
+    /// Updates the healthbar to display the remaining health as a fraction of the start health
+    /// </summary>
+    /// <param name="health">how much the health the enemy currently has</param>
+    public void UpdateHealthBar(float health) {
+        if (!healthBarBackground.activeSelf) { // if the health bar is not shown, show it.
+            DisplayHealthBar();
+        }
+        healthBar.fillAmount = health / startHealth;
```

Comment:
hvis du tester om helt<0  før du kjører da ungår du lag spikene man får nå med banan pistolen

Reply:
done. Litt overrasket over at du får spikes for jeg trodde CheckIfDead blir kjørt før denne.